aws_topology/stackstate_checks/aws_topology/resources/elb_classic.py

```diff
@@ -1,90 +1,151 @@
 import time
-from .utils import make_valid_data, create_resource_arn
+from .utils import (
+    client_array_operation,
+    make_valid_data,
+    create_arn as arn,
+    set_required_access_v2,
+    transformation,
+)
 from .registry import RegisteredResourceCollector
+from collections import namedtuple
+from schematics import Model
+from schematics.types import StringType, ListType, ModelType
 
 
 def create_arn(region=None, account_id=None, resource_id=None, **kwargs):
-    return "classic_elb_" + resource_id
+    return arn(
+        resource="elasticloadbalancing", region=region, account_id=account_id, resource_id="loadbalancer/" + resource_id
+    )
+
+
+LoadBalancerData = namedtuple("LoadBalancerData", ["elb", "tags", "instance_health"])
+
+
+class LoadBalancerInstances(Model):
+    InstanceId = StringType()
+
+
+class LoadBalancer(Model):
+    LoadBalancerName = StringType(required=True)
+    VPCId = StringType()
+    Instances = ListType(ModelType(LoadBalancerInstances))
+
+
+class InstanceHealth(Model):
+    InstanceId = StringType(required=True)
+    State = StringType()
+    Description = StringType()
 
 
 class ELBClassicCollector(RegisteredResourceCollector):
     API = "elb"
     API_TYPE = "regional"
     COMPONENT_TYPE = "aws.elb_classic"
     CLOUDFORMATION_TYPE = "AWS::ElasticLoadBalancing::LoadBalancer"
+    MAX_TAG_CALLS = 20  # This is the max items that can be requested in one describe_tags call
+
+    @set_required_access_v2("elasticloadbalancing:DescribeTags")
+    def collect_tag_page(self, elb_names):
+        max_items = self.MAX_TAG_CALLS  # Limit max items that we fetch to ensure call doesn't fail completely
+        return self.client.describe_tags(LoadBalancerNames=elb_names[:max_items]).get("TagDescriptions", [])
+
+    @set_required_access_v2("elasticloadbalancing:DescribeLoadBalancers")
+    def collect_elbs(self, **kwargs):
+        for elb in client_array_operation(self.client, "describe_load_balancers", "LoadBalancerDescriptions", **kwargs):
```

Comment:
Shouldn't the collect_instance_health be done in this loop?

Reply:
Does it need to be? I collect all resources in the same function together, process_elb_page

Reply:
Lets visually discuss. Can't see any more what I meant here :-)